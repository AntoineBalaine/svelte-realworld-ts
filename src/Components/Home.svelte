<script>
	import ArticlePreview from "./ArticlePreview.svelte";
	import TagSection from "./TagSection.svelte";
</script>

<div class="home-page">
	<div class="banner">
		<div class="container">
			<h1 class="logo-font">conduit</h1>
			<p>A place to share your knowledge.</p>
		</div>
	</div>

	<div class="container page">
		<div class="row">
			<div class="col-md-9">
				<div class="feed-toggle">
					<ul class="nav nav-pills outline-active">
						<li class="nav-item">
							<a class="nav-link disabled" href="">Your Feed</a>
						</li>
						<li class="nav-item">
							<a class="nav-link active" href="">Global Feed</a>
						</li>
					</ul>
				</div>

				<slot />

				<div class="article-preview">
					<div class="article-meta">
						<a href="profile.html"
							><img alt="profile thumbnail" src="http://i.imgur.com/N4VcUeJ.jpg" /></a
						>
						<div class="info">
							<a href="" class="author">Albert Pai</a>
							<span class="date">January 20th</span>
						</div>
						<button class="btn btn-outline-primary btn-sm pull-xs-right">
							<i class="ion-heart" /> 32
						</button>
					</div>
					<a href="" class="preview-link">
						<h1>The song you won't ever stop singing. No matter how hard you try.</h1>
						<p>This is the description for the post.</p>
						<span>Read more...</span>
					</a>
				</div>
			</div>

			<TagSection />
		</div>
	</div>
</div>
