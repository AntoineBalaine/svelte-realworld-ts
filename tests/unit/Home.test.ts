import { render } from "@testing-library/svelte";
import { describe, test, expect } from "vitest";
import Home from "src/components/Home.svelte";

import type { ComponentProps } from "svelte";
import { articleSample } from "tests/DataSamples";

describe("Home Component", () => {
	test("should render the component", () => {
		const homeProps: ComponentProps<Home> = {
			data: {
				articles: [articleSample],
				user: null,
				tags: { tags: Array<string>("a tag") },
				pages: 1,
				isLoggedIn: false,
				pageNumber: 1
			}
		};
		const homePreview = render(Home, homeProps);

		const links = homePreview.getAllByRole("link");
		expect(links[0].innerHTML).toEqual("Global Feed");
		expect(links[1].innerHTML).toEqual("Sign in to see your Feed");
		expect(homePreview.getByTestId("homepage-tag")).not.toBeUndefined();

		expect(homePreview.container).toMatchSnapshot();
	});
});
