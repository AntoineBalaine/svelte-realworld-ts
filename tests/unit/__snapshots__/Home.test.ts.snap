// Vitest Snapshot v1

exports[`VerticalTabs Component > should render the component 1`] = `
<body>
  <div>
    <div
      class="home-page"
    >
      <div
        class="banner"
      >
        <div
          class="container"
        >
          <h1
            class="logo-font"
          >
            conduit
          </h1>
           
          <p>
            A place to share your knowledge.
          </p>
        </div>
      </div>
       
      <div
        class="container page"
      >
        <div
          class="row"
        >
          <div
            class="col-md-9"
          >
            <div
              class="feed-toggle"
            >
              <ul
                class="nav nav-pills outline-active"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link active"
                    href="/"
                  >
                    Global Feed
                  </a>
                </li>
                 
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/login"
                  >
                    Sign in to see your Feed
                  </a>
                </li>
                 
              </ul>
            </div>
             
          </div>
           
          <div
            class="col-md-3"
          >
            <div
              class="sidebar"
            >
              <p>
                Popular Tags
              </p>
               
              <div
                class="tag-list"
              >
                <a
                  class="tag-pill tag-default"
                  data-testid="homepage-tag"
                  href="/?tag=a tag"
                >
                  a tag
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <!--&lt;Home&gt;-->
  </div>
</body>
`;
